htgr/htr-pm/tests

```diff
@@ -0,0 +1,77 @@
+[Tests]
+  [htr_pm_sam_main_syntax]
+    type = RunApp
+    input = 'ss-main.i'
+    check_input = True
+    executable_pattern = 'sam*|blue_crab*'
+    cli_args = '--app SamApp --allow-unused'
+  []
+  [htr_pm_sam_loop_syntax]
+    type = RunApp
+    input = 'ss-primary-loop-full.i'
+    check_input = True
+    executable_pattern = 'sam*|blue_crab*'
+    cli_args = '--app SamApp --allow-unused'
+  []
+  [htr_pm_sam_rccs_syntax]
+    type = RunApp
+    input = 'ss-rccs-water.i'
+    check_input = True
+    executable_pattern = 'sam*|blue_crab*'
+    cli_args = '--app SamApp --allow-unused'
+  []
+  [htr_pm_sam_main_transient_syntax]
+    type = RunApp
+    input = 'ss-main-transient.i'
+    check_input = True
+    executable_pattern = 'sam*|blue_crab*'
+    cli_args = '--app SamApp --allow-unused'
+  []
+  [htr_pm_sam_loop_transient_syntax]
+    type = RunApp
+    input = 'ss-primary-loop-transient.i'
+    check_input = True
+    executable_pattern = 'sam*|blue_crab*'
+    cli_args = '--app SamApp --allow-unused'
+  []
+  [htr_pm_sam_rccs_transient_syntax]
+    type = RunApp
+    input = 'ss-rccs-water-transient.i'
+    check_input = True
+    executable_pattern = 'sam*|blue_crab*'
+    cli_args = '--app SamApp --allow-unused'
+  []
+  [htr_pm_sam_main]
+    type = CSVDiff
+    input = 'ss-main.i'
+    csvdiff = 'ss-main_csv.csv'
+    executable_pattern = 'sam*|blue_crab*'
+    cli_args = '--app SamApp --allow-unused Executioner/dt=0.005 Executioner/num_steps=2 Executioner/nl_rel_tol=1e-8 Executioner/nl_abs_tol=1e-7'
+    abs_zero = 1e-5
+    rel_err = 1e-4
+    max_time = 1000
+    method = 'opt'
+  []
+  [htr_pm_sam_primary_loop]
+    type = CSVDiff
+    input = 'ss-primary-loop-full.i'
+    csvdiff = 'ss-primary-loop-full_csv.csv'
+    executable_pattern = 'sam*|blue_crab*'
+    cli_args = '--app SamApp --allow-unused Executioner/dt=0.005 Executioner/num_steps=2 Executioner/nl_rel_tol=1e-8 Executioner/nl_abs_tol=1e-7'
+    abs_zero = 1e-5
+    rel_err = 1e-4
+    max_time = 1000
+    method = 'opt'
+  []
+  [htr_pm_sam_rccs]
+    type = CSVDiff
+    input = 'ss-rccs-water.i'
+    csvdiff = 'ss-rccs-water_csv.csv'
+    executable_pattern = 'sam*|blue_crab*'
+    cli_args = '--app SamApp --allow-unused Executioner/dt=0.005 Executioner/num_steps=2 Executioner/nl_rel_tol=1e-8 Executioner/nl_abs_tol=1e-7'
+    abs_zero = 1e-5
+    rel_err = 1e-4
+    max_time = 1000
+    method = 'opt'
+  []
```

Comment:
we're missing regression tests for the transient input. Can we use an early checkpoint using cli_args to do this?

Reply:
Are the checkpoint files stored during testing? If they are stored, the transient tests can be restarted using the checkpoint files generated by the steady-state cases.

Reply:
they are generated and are not removed.

Reply:
you can find an example at msr/msre/reactivity_insertion

Reply:
we use the final checkpoint there iirc but if the steady simulation does not reach it in time, we can set hte number of time steps for the steady state

Reply:
added individual regression tests for all files